### PR TITLE
[Remove Stream/Optional usage] LPS-175655  info-collection-provider-item-selector-web

### PR DESCRIPTION
**Pull request by JanisZhang**
https://issues.liferay.com/browse/LPS-175655

### CODE REVIEW

**Comment by JanisZhang** [comment]
ci:stop

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by JanisZhang** [comment]
force-pushed. I added one SF commit

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by JanisZhang** [comment]
ci:test:item-selector

File: modules/apps/info/info-collection-provider-item-selector-web/src/main/java/com/liferay/info/collection/provider/item/selector/web/internal/item/selector/InfoCollectionProviderItemSelectorViewDescriptor.java [hunk 1]
```diff
@@ -197,17 +183,17 @@ private List<DropdownItem> _getFilterTypeDropdownItems() {
...
-			}
-		).distinct(
-		).sorted(
-			new KeyValuePairComparator(false, true)
-		).collect(
-			Collectors.toList()
-		);
+			});
+
+		ListUtil.distinct(keyValuePairs);
```
Comment (dantewang) [inline modules/apps/info/info-collection-provider-item-selector-web/src/main/java/com/liferay/info/collection/provider/item/selector/web/internal/item/selector/InfoCollectionProviderItemSelectorViewDescriptor.java:188]: @JanisZhang Please use 
```
ListUtil.distinct(keyValuePairs, new KeyValuePairComparator(false, true));
```
so you don't need to call `sort` separately.

**Comment by JanisZhang** [comment]
ci:stop

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by JanisZhang** [comment]
ci:test:item-selector

**Comment by JanisZhang** [comment]
retrigger ci:test:relevant

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
See https://github.com/liferay-core-infra/liferay-portal/pull/3718

**Pull request closed without merging**